examples/plugins/inspec-resource-lister/test/functional/inspec_resource_lister_test.rb

```diff
@@ -0,0 +1,110 @@
+  # import it.  Including it here will make it available in all child
+  # 'describe' blocks.
+  include CorePluginFunctionalHelper
+
+  # When thinking up scenarios to test, start with the simplest.
+  # Then think of each major feature, and exercise them.
+  # Running combinations of features makes sense if it is very likely,
+  # or a difficult / dangerous case.  You can always add more tests
+  # here as users find subtle problems.  In fact, having a user submit
+  # a PR that creates a failing functional test is a great way to
+  # capture the reproduction case.
+  # The simplest case:
+  describe "when run without an argument" do
+
+    # run_inspec_process_with_this_plugin is a helper provided by
+    # CoreFunctionalHelper. It makes the InSpec that Bundler installed
+    # think that this plugin we are currently testing is installed as a
+    # user plugin, by writing a plugin config file in a temp dir.
+    # To use it, just provide a command line, minus the word `inspec`.
+    let (:outcome) { run_inspec_process_with_this_plugin('listresources core') }
+
+    # Some tests through here use minitest Expectations, which attach to all
+    # Objects, and begin with 'must' (positive) or 'wont' (negative)
+    # See https://ruby-doc.org/stdlib-2.1.0/libdoc/minitest/rdoc/MiniTest/Expectations.html
+    it("should exit successfully") { outcome.exit_status.must_equal(0) }
+    it("should be silent on stderr") { outcome.stderr.must_be_empty }
+
+    # A selection of core resources, just spot checking.
+    # This is an example of using Ruby to define sets of tests.
+    ['process', 'service', 'user', 'file'].each do |resource_name|
+      it "should mention the '#{resource_name}' resource" do
+        outcome.stdout.must_include(resource_name)
+      end
+    end
+
+    # Check for the summary
+    it "should mention the summary" do
+      outcome.stdout.must_include('resources total')
+    end
+  end
+
+  # Test the search pattern feature, in a couple of ways.
+  describe "when run with a search pattern that matches things" do
+    # Notice that the command line is changed here:
+    # "list all resources that have the word user in them"
+    let (:outcome) { run_inspec_process_with_this_plugin('listresources core user') }
+
+    # Should be well-behaved...
+    it("should exit successfully") { outcome.exit_status.must_equal(0) }
+    it("should be silent on stderr") { outcome.stderr.must_be_empty }
+
+    # Here, we want to know it DID match some things, and NOT some others.
+    ['user', 'users'].each do |resource_name|
+      it "should mention the '#{resource_name}' resource" do
+        outcome.stdout.must_include(resource_name)
+      end
+    end
+    ['process', 'service', 'file'].each do |resource_name|
+      it "should NOT mention the '#{resource_name}' resource" do
+        outcome.stdout.wont_include(resource_name)
+      end
+    end
+  end
+  describe "when run with a search pattern that matches nothing" do
```

Comment:
There are newlines between blocks elsewhere in the file. I'll leave it to you though.